test/dogma/rules_test.exs

```diff
@@ -0,0 +1,79 @@
+defmodule Dogma.RulesTest do
+  use ExUnit.Case, async: false
+  import Mock
+
+  alias Dogma.Rules
+  alias Dogma.Script
+  alias Dogma.Test.AssertingFormatter
+
+  @formater_for_testing Dogma.Formatter.Simple
+
+  test "returns empty list when given no scripts" do
+    result = Rules.test([], @formater_for_testing)
+    assert result == []
+  end
+
+  test "Updates a script with the errors from rules" do
+    with_mock Dogma.RuleSet.All, [rules: fn() -> [{FakeRule}] end] do
+      single_script = %Script{}
+      expected_script_after_run = %Script{errors: [:always_fake_error]}
+
+      result = Rules.test([single_script], @formater_for_testing)
+
+      assert result == [expected_script_after_run]
+    end
+  end
+
+  test "Config can be used to define the rules used" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
+    single_script = %Script{}
+    expected_script_after_run = %Script{errors: [:always_fake_error]}
+
+    result = Rules.test([single_script], @formater_for_testing)
+
+    assert result == [expected_script_after_run]
+  end
+
+  test "Formatter.script is called by Rules.test" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
+    single_script = %Script{}
+    script_after_run = %Script{errors: [:always_fake_error]}
+
+    AssertingFormatter.start_listening
+    Rules.test([single_script], AssertingFormatter)
+
+    AssertingFormatter.assert_script_called_with script_after_run
+  end
+
+end
+
+defmodule Dogma.RuleSets.FakeRuleSet do
```

Comment:
Why wouldn't that work, sorry?

I see your point, but I don't like having to use fully qualified module names in the config file. Perhaps we should check to see if it is defined under `Dogma.Rule`, first, and then treat it as a full name if not.